src/browser-connection/local.js

```diff
@@ -28,11 +29,29 @@ export default class LocalBrowserConnection extends BrowserConnection {
         });
     }
 
+    _waitForClose () {
+        return new Promise(resolve => {
+            if (this.disconnected) {
+                resolve();
+                return;
+            }
+
+            var waitingTimeout = setTimeout(() => {
+                resolve();
+            }, this.WAITING_FOR_CLOSE_TIMEOUT);
+
+            this.once('closed', () => {
+                clearTimeout(waitingTimeout);
+                resolve();
+            });
+        });
+    }
 
-    close () {
+    async close () {
```

Comment:
I believe we shouldn't make `close` async for two reasons:
 1. It involves incosisteny - base `close` is not actually async, so the child shouldn't. Otherwise we break [LSP](https://en.wikipedia.org/wiki/Liskov_substitution_principle)
 2. We have a weird confusing logic right now, self-subscribing to own `close` event and then resolving close.

How I see it:
Local connection should have sync `close` method and `closed` event. When `close()` called it's just sets the `forceClose` and initiates timer. On the other hand in `getStatus` we await browser natives `closeBrowser`, call `super.close()` and emit `closed` event. If event doesn't occur in the time given by timer which was set in `close()`, we force `closed` event.